deployments/puppet/README.md

```diff
@@ -30,24 +43,24 @@ This class accepts the following parameters:
 
 | Name | Description | Default value |
 | :--- | :---------- | :------------ |
+| `collector_version` | **Required on Windows**: Version of the collector package to install, e.g., `0.25.0`.  The version should correspond to [Github Releases](https://github.com/signalfx/splunk-otel-collector/releases) _without_ the preceding `v`.  **Note**: On Linux, the latest collector version will be installed if this parameter is not specified. | None |
```

Comment:
> Is this not something we can provide a default for to make optional?

The logic was migrated from the smart agent (https://github.com/signalfx/signalfx-agent/blob/main/deployments/puppet/manifests/init.pp#L28-L31),  but I can try to default to latest.

Reply:
AFAICT, I think we made it required on windows for the smart agent because there is no good way in puppet to determine the latest version and dynamically set the version var at runtime (on linux, apt and yum just handles the latest version).  Thoughts?

Reply:
I may not be following but is reading from `https://dl.signalfx.com/splunk-otel-collector/msi/release/latest.txt` not an option if the version isn't set?

Reply:
The issue seems to be that puppet variable values need to be set at compile time, so not yet sure how to get the version from `https://dl.signalfx.com/splunk-otel-collector/msi/release/latest.txt` and download the MSI based on this version at runtime. But I can continue to investigate for a later release.